run.cc

```diff
@@ -68,12 +73,60 @@ int main() {
   cout << "******************************************************" << endl;
   // create an object which holds data of experiment A
   Data datA("exp_A");
+  Data datB("exp_B");
+  Data datC("exp_C");
+  Data datD("exp_D");
+  std::vector<Data> data;
+  data.push_back(datA);
+  data.push_back(datB);
+  data.push_back(datC);
+  data.push_back(datD);
+  
+ 
+
 
   // here is the data from experiment A
   cout << "bin 27: from " << datA.binLow(27) << " to " << datA.binHigh(27)
        << endl;
   cout << "measurement of experiment A in bin 27: " << datA.measurement(27)
        << endl;
+  cout << "measurement of experiment B in bin 27: " << datB.measurement(27)
+       << endl;
+  cout << "measurement of experiment C in bin 27: " << datC.measurement(27)
+       << endl;
+  cout << "measurement of experiment D in bin 27: " << datD.measurement(27)
+       << endl;
+
+  double uncert_A = datA.error(27);
+  double uncert_B = datB.error(27);
+
+  cout << "n*uncertainty " << 2*sqrt(pow(uncert_A,2)+pow(uncert_B,2)) << endl;
+  cout << "delta Y " << abs(datA.measurement(27)-datB.measurement(27)) << endl;
+
+  int count = datA.checkCompatibility(datD, 1);
+  cout << "count: " << count << endl;
+  Data datAB = datA.averageDatasets(datB);
+  Data datCD = datC.averageDatasets(datD);
+  Data datABCD = datAB.averageDatasets(datCD);
+  
+  cout << "for datA, " << datA.checkCompatibility(datA, 2) << " datapoints differ more than 2 std's" << endl;
+  cout << "for datB, " << datB.checkCompatibility(datA, 2) << " datapoints differ more than 2 std's" << endl;
+  cout << "for datC, " << datC.checkCompatibility(datA, 2) << " datapoints differ more than 2 std's" << endl;
+  cout << "for datD, " << datD.checkCompatibility(datA, 2) << " datapoints differ more than 2 std's" << endl;
+
+  cout << "for datA, " << datA.checkCompatibility(datA, 3) << " datapoints differ more than 3 std's" << endl;
+  cout << "for datB, " << datB.checkCompatibility(datA, 3) << " datapoints differ more than 3 std's" << endl;
+  cout << "for datC, " << datC.checkCompatibility(datA, 3) << " datapoints differ more than 3 std's" << endl;
+  cout << "for datD, " << datD.checkCompatibility(datA, 3) << " datapoints differ more than 3 std's" << endl;
+
+
+  
+  double chi_DatA = datA.chi2perNdf();
+  cout << "chi2 DatA: " << datA.chi2perNdf() << endl;
+  cout << "chi2 DatB: " << datB.chi2perNdf() << endl;
+  cout << "chi2 DatC: " << datC.chi2perNdf() << endl;
+  cout << "chi2 DatD: " << datD.chi2perNdf() << endl;
+  cout << "chi2 DatABCD: " << datABCD.chi2perNdf() << endl;
 
```

Comment:
You didn't answer if the data are compatible with the model using the given parameters. The chisquare values you printed out are correct, but do you think they are 'good'? What should the value ideally be and are the deviations from that value that you see here significant? -0.5p